create-custom-map-style/README.md

```diff
@@ -0,0 +1,823 @@
+
+ ![App Skeleton](media/AppSkeleton.png)
+ *App Skeleton*
+
+### React - Amazon Location Service Integration
+
+1. Now it's time for adding yet another package; this time a very important library that uses WebGL to render interactive maps from [vector tiles](https://docs.mapbox.com/help/glossary/vector-tiles/) and [Maplibre styles](https://maplibre.org/maplibre-gl-js-docs/style-spec/)
+
+    ```bash
+
+    npm install maplibre-gl --save
+    ```
+
+1. With this we'll start adding the map to our page. Start by importing the dependencies in your `SamplePage` component
+
+    ```js
+
+    ...
+    import maplibregl from 'maplibre-gl';
+    ...
+    ```
+
+1. Here's where we'll get a chance to fire up our react hooks. We need to add a reference (`useRef`) to the `.demo-map` DOM node, make an asynchronous API call (`useEffect`) to fetch data using Amplify, and keep track of the latitude, longitude, and zoom level of the map (`useState`)
+
+    ```js
+
+    ...
+    import  React, { useRef, useState, useEffect } from  'react';
+    ...
+    import { createMap } from "maplibre-gl-js-amplify";
+    import style from './example-style-descriptor.json';
+    ...
+    const SamplePage = () => {
+      const map = useRef();
+
+      const [coordinates, setCoordinates] = useState({
+        lat: 40.7356,
+        lng: -74.0541,
+        zoom: 12.44,
+      });
+
+      useEffect(async () => {
+        const { lat, lng, zoom } = coordinates;
+        const demoMap = await createMap({
+          container: map.current,
+          center: [lng, lat],
+          zoom,
+          style,
+        });
+        demoMap.on('move', () => {
+          setCoordinates({
+            lng: demoMap.getCenter().lng.toFixed(4),
+            lat: demoMap.getCenter().lat.toFixed(4),
+            zoom: demoMap.getZoom().toFixed(2),
+          });
+        });
+        return () => demoMap.remove();
+      }, []);
+    ...
+
+    // And in your return statement below change
+    // <div className="demo-map" />
+    // to
+
+    <div ref={map} className="demo-map" />
+    ```
+
+1. `maplibre-gl-js-amplify` takes care of authenticating our user before calling Location Service when fetching the map tiles.
+
+1. You should be able to go back to your browser now and see the base map style show up inside your `demo-map` div
+
+ ![Esri Streets Base Map](media/BaseMap.png)
+ *Esri Streets Base Map*
+
+## Styling a Custom Map
+
+1. Maputnik is an open source visual editor for the [Mapbox Style Specification](https://www.mapbox.com/mapbox-gl-js/style-spec). At the time of writing this editor can be either used in the browser or downloaded as a binary and can run locally. We are going to use the latter option.
+
+1. Go to <https://github.com/maputnik/editor/wiki/Maputnik-CLI> and follow the instructions for installation or optionally from your terminal:
+
+    ```bash
+
+    wget <https://github.com/maputnik/editor/releases/download/v1.7.0/maputnik-darwin.zip>
```

Comment:
This is Mac-specific; that's probably fine, but include that in the opening to avoid disappointing readers on other platforms after they've gone through all of the Amplify stuff.

Reply:
The `<>`s will also throw off readers who follow these steps verbatim.